### PR TITLE
chore: add data partition mount check in systemd

### PR DESCRIPTION
**Pull request by snusifer**

### CODE REVIEW

File: mender-convert-modify [hunk 1]
```diff
@@ -164,6 +164,28 @@ if [ "${MENDER_GRUB_EFI_INTEGRATION}" == "y" ]; then
 
 fi
 
+cat << EOF > /work/mender-client-data-dir.service
```
Comment (oleorhagen) [inline mender-convert-modify:167]: Maybe we can get the file from `meta-mender` ? :)

File: mender-convert-modify [hunk 2]
```diff
@@ -164,6 +164,16 @@ if [ "${MENDER_GRUB_EFI_INTEGRATION}" == "y" ]; then
 
 fi
 
+run_and_log_cmd "sudo git init work/meta-mender"
+run_and_log_cmd "sudo git -C work/meta-mender remote add origin https://github.com/mendersoftware/meta-mender.git"
+run_and_log_cmd "sudo git -C work/meta-mender fetch --depth 1 origin e05cde10fde3646a78081709e717fb42a6c5ba44"
```
Comment (snusifer) [inline mender-convert-modify:169]: Q:
Why not get this file with `curl` against github? :thinking:

Comment (oleorhagen) [reply mender-convert-modify:169]: ?

Comment (snusifer) [reply mender-convert-modify:169]: Ah, nvm

Comment (snusifer) [reply mender-convert-modify:169]: Could curl it from https://raw.githubusercontent.com/mendersoftware/meta-mender/master/meta-mender-core/recipes-mender/mender-client/files/mender-client-data-dir.service
But no guarantee that url won't change in the future

Comment (oleorhagen) [reply mender-convert-modify:169]: Yes, I agree :)

File: mender-convert-modify [hunk 3]
```diff
@@ -164,6 +164,15 @@ if [ "${MENDER_GRUB_EFI_INTEGRATION}" == "y" ]; then
 
 fi
 
+MENDER_CLIENT_DATA_DIR_SERVICE_URL="https://raw.githubusercontent.com/mendersoftware/\
+meta-mender/e05cde10fde3646a78081709e717fb42a6c5ba44/meta-mender-core/recipes-mender/mender-client/files/mender-client-data-dir.service"
+
+run_and_log_cmd "wget --quiet -O work/mender-client-data-dir.service $MENDER_CLIENT_DATA_DIR_SERVICE_URL"
+run_and_log_cmd "sudo install -m 644 work/mender-client-data-dir.service work/rootfs/lib/systemd/system/mender-client-data-dir.service"
+run_and_log_cmd "sudo mkdir -p work/rootfs/lib/systemd/system/mender-client.service.wants"
+run_and_log_cmd "sudo ln -sf work/rootfs/lib/systemd/system/mender-client-data-dir.service \
```
Comment (oleorhagen) [inline mender-convert-modify:173]: @snusifer have you tested this? Will not the symlink contain the `work` prefix here? :thinking:

Comment (snusifer) [reply mender-convert-modify]: woops

File: mender-convert-modify [hunk 4]
```diff
@@ -164,6 +164,15 @@ if [ "${MENDER_GRUB_EFI_INTEGRATION}" == "y" ]; then
 
 fi
 
+MENDER_CLIENT_DATA_DIR_SERVICE_URL="https://raw.githubusercontent.com/mendersoftware/\
+meta-mender/e05cde10fde3646a78081709e717fb42a6c5ba44/meta-mender-core/recipes-mender/mender-client/files/mender-client-data-dir.service"
+
+run_and_log_cmd "wget --quiet -O work/mender-client-data-dir.service $MENDER_CLIENT_DATA_DIR_SERVICE_URL"
+run_and_log_cmd "sudo install -m 644 work/mender-client-data-dir.service work/rootfs/lib/systemd/system/mender-client-data-dir.service"
+run_and_log_cmd "sudo mkdir -p work/rootfs/lib/systemd/system/mender-client.service.wants"
+run_and_log_cmd "sudo ln -sf ../mender-client-data-dir.service \
```
Comment (oleorhagen) [inline mender-convert-modify:173]: I would suggest using the absolute path for the symlink @snusifer  :)

**Review by oleorhagen — APPROVED** [review]

**Pull request merged**